library/Vo/DateRange.php

```diff
@@ -476,4 +476,34 @@ public function __toString()
             )
         );
     }
+
+    /**
+     * Determine if the range is anchored in the future.
+     *
+     * @return bool
+     */
+    public function isFuture()
+    {
+        return new DateTime(static::FUTURE) == $this->getEnd();
+    }
+
+    /**
+     * Determine if the range is anchored in the past.
+     *
+     * @return bool
+     */
+    public function isPast()
+    {
+        return new DateTime(static::PAST) == $this->getStart();
+    }
+
+    /**
+     * Determine if the range is infinite.
+     *
+     * @return bool
+     */
+    public function isInfinite()
+    {
+        return self::infinite() == $this;
```

Comment:
I think this should be `static::` instead of `self::`, in case `infinite()` is overridden by a subclass. It's also consistent with the `isFuture()` and `isPast()`.